cf_units/_udunits2_parser/udunits2Parser.g4

```diff
@@ -1,6 +1,79 @@
 parser grammar udunits2Parser;
-  
+
 options { tokenVocab=udunits2Lexer; }
 
 
-unit_spec: ALL;
+unit_spec:
+    shift_spec? EOF  // Zero or one "shift_spec", followed by the end of the input.
+;
+
+shift_spec:
+    product
+    | product shift number    // Kelvin @ 273.15
+    | product shift timestamp // hours since 2001-12-31 23:59:59.999 +6
+;
+
+product:
+    power
+    | product power             // e.g. m2s (s*m^2)
+    | product MULTIPLY power    // e.g. m2*s
+    | product DIVIDE power      // e.g. m2/2
+    | product WS+ power         // e.g. "m2 s"
+;
+
+power:
+    basic_spec integer   // e.g. m+2, m2. Note that this occurs *before* basic_spec,
+                         // as m2 should be matched before m for precendence of power
+                         // being greater than multiplication (e.g. m2==m^2, not m*2).
+    | basic_spec
+    | basic_spec RAISE integer      // e.g. m^2
+    | basic_spec UNICODE_EXPONENT   // e.g. m²
+;
+
+basic_spec:
+    ID
+    | '(' shift_spec ')'
+//    | LOGREF product_spec ')'
```

Comment:
@pelson Add a comment TBD or raise and issue? A nice simple extension for someone to follow-up on and get stuck into the parser...